MINOR: socket setup max should be 30 seconds

cmccabe opened the pull request:
The default socket.connection.setup.timeout.max.ms should be 30 seconds.
The current value of 127 seconds is longer than the default API timeout
for AdminClient, and longer than the default request timeouts for the
producer and consumer.  We should bring these configs into line with
each other.

hachikuji commented on clients/src/main/java/org/apache/kafka/clients/admin/AdminClientConfig.java:
```diff
@@ -162,7 +162,7 @@
                                         CommonClientConfigs.SOCKET_CONNECTION_SETUP_TIMEOUT_MS_DOC)
                                 .define(SOCKET_CONNECTION_SETUP_TIMEOUT_MAX_MS_CONFIG,
                                         Type.LONG,
-                                        CommonClientConfigs.DEFAULT_SOCKET_CONNECTION_SETUP_TIMEOUT_MAX_MS,
```
I think the producer and consumer have the same problem. Maybe we can just change the constant?

cmccabe replied:
seems reasonable.

cmccabe committed:
MINOR: socket setup max should be 30 seconds #10306

socket.connection.setup.timeout.max.ms should be 30 seconds.
The current value of 127 seconds is longer than the default API timeout
for AdminClient, and longer than the default request timeouts for the
producer and consumer.  We should bring these configs into line with
each other.

hachikuji commented:
cc @vvcephei This might be a good one for 2.8 since it is a simple fix.

cmccabe commented:
I retried the tests that failed locally and they all worked for me.  Will commit in a minute.

cmccabe commented:
Committed to trunk and 2.8

ijuma commented:
@cmccabe We should update the KIP and share a note in the mailing list thread. Also, was this released with `127` as the default or did we change it before the first release? If the former, we should add a release note, right?
